akka-actor-tests/src/test/scala/akka/util/TokenBucketSpec.scala

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.util
+
+import akka.testkit.AkkaSpec
+
+import scala.util.Random
+
+class TokenBucketSpec extends AkkaSpec {
+
+  class TestBucket(_cap: Long, _period: Long) extends TokenBucket(_cap, _period) {
+    var currentTime: Long = 0L
+  }
+
+  "A Token Bucket" must {
```

Comment:
The kind of laws that ScalaCheck can express are very limiting. The laws here are expressed in the last test which expresses a behavioral equality (bi-simulation if you like) between the IUT and a reference model. ScalaCheck rather ugly in these kind of tests: https://github.com/rickynils/scalacheck/blob/master/doc/UserGuide.md#stateful-testing